src/matrixrtc/EncryptionManager.ts

```diff
@@ -0,0 +1,500 @@
+import { type MatrixClient } from "../client.ts";
+import { logger as rootLogger } from "../logger.ts";
+import { MatrixEvent } from "../models/event.ts";
+import { Room } from "../models/room.ts";
+import { EncryptionConfig } from "./MatrixRTCSession.ts";
+import { secureRandomBase64Url } from "../randomstring.ts";
+import { EncryptionKeysEventContent } from "./types.ts";
+import { decodeBase64, encodeUnpaddedBase64 } from "../base64.ts";
+import { MatrixError, safeGetRetryAfterMs } from "../http-api/errors.ts";
+import { CallMembership } from "./CallMembership.ts";
+import { EventType } from "../@types/event.ts";
+const logger = rootLogger.getChild("MatrixRTCSession");
+
+/**
+ * A type collecting call encryption statistics for a session.
+ */
+export type Statistics = {
+    counters: {
+        roomEventEncryptionKeysSent: number;
+        roomEventEncryptionKeysReceived: number;
+    };
+    totals: {
+        roomEventEncryptionKeysReceivedTotalAge: number;
```

Comment:
```suggestion
        /**
         * The total age (in milliseconds) of all room events containing encryption keys that we have received.
         * We track the total age so that we can later calculate the average age of all keys received.
         */
        roomEventEncryptionKeysReceivedTotalAge: number;
```